### PR TITLE
HDDS-9777. Migrate simple filesystem integration tests to JUnit5

### PR DESCRIPTION
**Pull request by raju-balpande**
## What changes were proposed in this pull request?
Migrate the following test classes to JUnit5. See parent task for details.
`
- hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestLeaseRecovery.java
- hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFSInputStream.java
- hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFSWithObjectStoreCreate.java
- hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemMetrics.java
- hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemMissingParent.java
- hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemPrefixParser.java
- hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsHAURLs.java
- hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedDDSWithFSO.java
`

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9777

## How was this patch tested?
Tested on CI build. There was one failure 
`TestBackgroundContainerDataScannerIntegration.testCorruptionDetected()`
which I tested as a flacky test and it worked correctly https://github.com/raju-balpande/apache_ozone/actions/runs/7168012018

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsHAURLs.java [hunk 1]
```diff
@@ -299,14 +296,14 @@ public void testWithQualifiedDefaultFS() throws Exception {
...
-        Assert.assertThat("Command did not print the error message " +
+        MatcherAssert.assertThat("Command did not print the error message " +
                 "correctly for test case: "
                 + "ozone fs -ls o3fs://bucket.volume.id1:port/",
             capture.getOutput(), StringContains.containsString(
                 "does not use port information"));
       }
       // Check return value, should be -1 (failure)
-      Assert.assertEquals(res, -1);
+      assertEquals(-1, res);
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsHAURLs.java:306]: Nice! (swapping arguments to reflect `expected, actual` order)

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @raju-balpande for the patch.  Two classes have leftover JUnit4 assertions, but otherwise LGTM.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestLeaseRecovery.java
[the remark this answers is not in the document]
Comment (raju-balpande) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestLeaseRecovery.java:1]: Thanks, Done.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemMissingParent.java
[the remark this answers is not in the document]
Comment (raju-balpande) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemMissingParent.java:1]: Thanks, Done.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsHAURLs.java [hunk 2]
```diff
@@ -41,36 +41,34 @@
...
 import static org.apache.hadoop.hdds.HddsUtils.getHostName;
 import static org.apache.hadoop.hdds.HddsUtils.getHostPort;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 /**
  * Test client-side URI handling with Ozone Manager HA.
  */
+@org.junit.jupiter.api.Timeout(300)
```
[the remark this answers is not in the document]
Comment (raju-balpande) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsHAURLs.java]: Thanks, Done.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsHAURLs.java (hunk 1 above)
Comment (raju-balpande) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsHAURLs.java:307]: Thanks.

**Pull request merged**